test/e2e-common.sh

```diff
@@ -370,7 +370,17 @@ function run_postinstall_jobs() {
 function uninstall_channel_crds() {
   echo "Uninstalling Kafka Channel CRD"
   kubectl delete secret -n "${SYSTEM_NAMESPACE}" kafka-cluster
-  sleep 10 # Give Controller Time To React To Kafka Secret Deletion ; )
+
+  # The distributed channel controller no longer actively monitors the kafka-cluster secret,
+  # so the receiver will only be torn down if there is a change to the kafkachannels or the
+  # config-kafka configmap.  However, if the secret is missing, the global resync does not happen
+  # (the controller presumes this missing secret is an error and reacts accordingly), so this
+  # manual deletion of the receiver deployment is the easiest solution here.
+  if [[ $1 == "distributed" ]]; then
+    kubectl delete deployment -n "${SYSTEM_NAMESPACE}" kafka-cluster-32603413-receiver
+    kubectl delete service -n "${SYSTEM_NAMESPACE}" kafka-cluster-32603413-receiver
```

Comment:
`32603413` is there a deterministic process to generate this number or is this just forgotten?

Reply:
Yes, this comes from GenerateHash() in pkg/channel/distributed/controller/util/hash.go
For a secret named "kafka-cluster" the hash will always be 32603413
We could replicate the GenerateHash function in shell script here if that's more understandable; it would look something like this but with a parameter
```
echo -n "kafka-cluster" | md5 | sed 's/^\(........\).*/\1/g'
```

Reply:
We are planning, however (with issue #608), to use a non-hash-decorated single-name for the receiver, so I don't think adding this hash function (simple though it is) to the bash script will really be a long-term gain in usefulness.  (It does need to stay in the go code; the same function is used for the dispatcher names -- those pods are removed automatically when the kafkachannel is deleted during the test so a similar kubectl command isn't needed here)

Reply:
This ended up being such a small change that I opted for the visibility of adding the hash generation and a comment about the reason for the value, even if it ends up being temporary.

Reply:
Thanks. I am happy with both as long as there's determinism :)